concepts/0017-attachments/README.md

```diff
@@ -0,0 +1,466 @@
+to a communication context. [Messages are a mechanism whereby
+state evolves in a protocol](../0003-protocols/README.md#ingredients).
+Protocols are [versioned according to the structure and semantics of
+messages](../0003-protocols/README.md#semver-rules).
+Messages are usually small, consisting of a modest number of fields with
+a structure that's focused on furthering the goals of their protocol.
+
+* _Data has meaning at rest_, in many different DIDComm protocols, or
+in important contexts beyond DIDComm. Data may be very large and very
+complex. It may come in formats that are quite independent from
+DIDComm. Data may be produced, consumed or handled as part of a
+protocol, but the actual content of the data is usually not where
+processing at the protocol level focuses. In agent codebases, it would
+be common for data handling to be implemented in different classes
+or libraries from the handlers for messages.
+
+Some examples:
+
+* A protocol to negotiate the release of medical records might cause
+X-Rays, genomes, and many other artifacts to be transmitted. These
+artifacts are data, whereas the information packets that arrange the
+transmission and provide a carrying mechanism for the artifacts are
+messages.
+
+* A DIDComm message can be used to [report an error](https://github.com/hyperledger/indy-hipe/blob/d6503aeb/text/error-reporting/README.md). Descriptive
+parameters that change how the error is processed are probably
+part of the message, whereas a log file that provides supporting
+information should be thought of as data rather than the message
+proper. 
+
+* The protocol for issuing credentials consists of messages that flow
+through certain steps. One of the steps eventually delivers a credential.
+The credential is _data_; it has meaning even when the protocol is
+complete, and the protocol version may evolve independent of the data
+format of the credential itself. The fact that the credential is transmitted
+through a message does not change the credential's primary status as
+data.
+
+* A protocol to schedule a venue for an event might produce a confirmation
+message when it finishes. This message might include a map of the
+venue, instructions about how to unlock the gate, pictures of certain
+resources, and so forth. This collateral is _data_, whereas the messages
+that signal progression through the steps of scheduling are not.
+
+* The [Connection Protocol](https://github.com/hyperledger/indy-hipe/blob/master/text/0031-connection-protocol/README.md)
+exchanges messages to establish a connection between two parties. Part of
+what's exchanged is a DID Doc. The DID Doc is more like _data_ than it is
+like an ordinary _message_, since it has meaning at rest and outside the
+protocol.
+
+The line between these two concepts may not be perfectly crisp in all cases,
+and that is okay. It is clear enough, most of the time, to provide context
+for the central question of this HIPE, which is:
```

Comment:
This should be RFC. Other than that it looks good. I'm going to merge and then we can PR update this later.